qiskit/primitives/base_estimator.py

```diff
@@ -227,12 +259,104 @@ def __call__(
             values = parameter_values[i].
 
         Args:
-            circuit_indices: the list of circuit indices.
-            observable_indices: the list of observable indices.
+            circuits: the list of circuit indices.
+            observables: the list of observable indices.
             parameter_values: concrete parameters to be bound.
             run_options: runtime options used for circuit execution.
 
         Returns:
             EstimatorResult: The result of the estimator.
+
+        Raises:
+            QiskitError: For mismatch of object id.
+            QiskitError: For mismatch of length of Sequence.
         """
+
+        # Support ndarray
+        if isinstance(parameter_values, np.ndarray):
+            parameter_values = parameter_values.tolist()
+
+        # Allow lift T to [T].
+        if circuits is not None and not isinstance(circuits, (np.ndarray, Sequence)):
+            circuits = [circuits]
+        if observables is not None and not isinstance(observables, (np.ndarray, Sequence)):
+            observables = [observables]
+        if parameter_values is not None and not isinstance(
+            parameter_values[0], (np.ndarray, Sequence)
+        ):
+            parameter_values = cast("Sequence[float]", parameter_values)
+            parameter_values = [parameter_values]
+
+        # Allow broadcasting
+        if (
+            circuits is None
+            and len(self._circuits) == 1
+            and observables is None
+            and len(self._observables) == 1
+            and parameter_values is not None
+        ):
+            circuits = [0] * len(parameter_values)
+            observables = [0] * len(parameter_values)
+
+        # Allow optional
+        if circuits is None:
+            circuits = list(range(len(self._circuits)))
+        if observables is None:
+            observables = list(range(len(self._observables)))
+        if parameter_values is None:
+            parameter_values = [[]] * len(circuits)
```

Comment:
```suggestion
        if parameter_values is None:
            for i in circuits:
                if len(self._circuits[i].parameters) != 0:
                    raise QiskitError(
                        f"The {i}-th circuit ({len(circuits)}) is parameterised,"
                        "but parameter values are not given."
                    )
            parameter_values = [[]] * len(circuits)
```
Do we want to raise this error here like in the internal implementation?

Reply:
Thanks a lot. It is certainly helpful to have this error message.